source/run-command.txt

```diff
@@ -0,0 +1,188 @@
+   
+   To perform administrative tasks, use the :mongosh:`MongoDB Shell </>`
+   instead of the {+php-library+}. The shell provides helper methods
+   that might not be available in the library.
+   
+   If there are no available helpers in the library or the shell, you
+   can use the ``db.runCommand()`` shell method.
+
+.. _php-execute-command:
+
+Execute a Command
+-----------------
+
+To run a database command, you must specify the command and any relevant
+parameters in a command document, then pass the command document to the
+``MongoDB\Database::command()`` method. Many database commands return
+multiple result documents, so the ``command()`` method returns a
+``Cursor`` object that you can iterate through.
+
+The following code shows how you can use the ``command()``
+method on a :phpclass:`MongoDB\Database` instance to run the ``hello``
+command, which returns information about the current member's role in
+the replica set:
+
+.. literalinclude:: /includes/write/run-command.php
+   :language: php
+   :dedent:
+   :start-after: start-hello
+   :end-before: end-hello
+
+To find a link to a full list of database commands and corresponding
+parameters, see the :ref:`Additional Information section
+<php-addtl-info-runcommand>`.
+
+.. note:: Read Preference
+
+   The ``command()`` method does not obey the read preference you might
+   have set on your ``Database`` instance elsewhere in your code. By
+   default, ``command()`` uses the ``primary`` read preference.
+   
+   You can set a read preference for command execution by setting one
+   in the options parameter, as shown in the following code:
+   
+   .. literalinclude:: /includes/write/run-command.php
+      :language: php
+      :dedent:
+      :start-after: start-readpref
+      :end-before: end-readpref
+
+   For more information on read preference options, see :manual:`Read
+   Preference </core/read-preference/>` in the {+mdb-server+} manual.
+
+.. _php-command-response:
+
+Response
+--------
+
+The ``command()`` method returns a ``Cursor`` object that contains
+the response from the database for the given command. Each database
+command performs a different function, so the response
+content can vary. Some command responses contain multiple result
+documents. In these situations, the library converts the cursor
+envelope in the raw command response, which includes the cursor ID and
+the first batch of results, into an iterable cursor.
+
+Before you run a command, learn about the response format of the
+command so that your application either iterates through multiple
+results or extracts the first and only document in the cursor. See the
+:ref:`php-addtl-info-runcommand` section of this guide to find a link to
+the full list of database commands.
+
+If the number of documents in the command response is sufficiently large, you
+can run a :manual:`getMore </reference/command/getMore/>` command to
+retrieve the next batch of results from the cursor by using the cursor
+ID.
```

Comment:
I see - this applies only for the shell method I guess? Misunderstood a comment you made in the previous review.

Reply:
The behavior in PHPLIB's `Database::command()` (and PHPC's [`executeCommand()`](https://www.php.net/manual/en/mongodb-driver-manager.executecommand.php)) method is probably unique. Other drivers may have a separate method for the envelope handling (see: [runCursorCommand](https://github.com/mongodb/specifications/blob/master/source/run-command/run-command.md#runcursorcommand) in the driver spec).

AFAIK, [`db.runCommand()`](https://www.mongodb.com/docs/manual/reference/method/db.runCommand/) returns the raw response with no envelope unpacking, and I don't think the shell has a command cursor variant. I'm also not aware of any way in the shell to construct a cursor object from a raw response, so a script written for mongosh would probably need to manually execute `getMore` commands.

----

But more generally, if you have a cursor object in the driver or shell, you can trust that it will use `getMore` as needed to continue iteration. And `db.runCommand()` does not give you a cursor.